lms/models/_mixins/public_id.py

```diff
@@ -36,3 +36,26 @@ def public_id(self, region: Region) -> str:
                 instance_id=self._public_id,
             )
         )
+
+    @classmethod
+    def public_id_eq(cls, public_id, region: Region):
```

Comment:
I find the comparator to be a bit confusing, I tend to prefer to leave the query side of SQLA as vanilla as possible, column on the left, value on the right.

I rather expose the parsed PublicID here and then just do `_public_id==model.parsed_public_id.instance_id` on the service side.

We'd get `model.parsed_public_id.region|app_code|...` for free which could be useful and more convenient than having another usage of `PublicId.parse`

`parsed_public_id` is not the best name ever.